feat(xx1_gen2): update XX1 gen2 sensor configurations

KYabuuchi opened the pull request:
## Description
This is a PR for configuration changes for XX1 Gen2.  It modifies the **GNSS**, **LiDAR**, and **Radar** settings for XX1 Gen2.

> [!NOTE]
> Other sensor_models are not affected. :ok_hand: 

## Changes to LiDAR

1. In `lidar_gen2.yaml` , the max_range values have been updated, and unnecessary parameters have been removed:

[TIER IV INTERNAL LINK](https://star4.slack.com/archives/CEV8XMJBV/p1731407529844719)

|lidar name| updated max range|
|-|-|
|top| 300m|
|front_left |80m|
|front_right|80m|
|side_left  |10m|
|side_right |10m|

2. Updated `lidar.launch.py` to set the return_mode to `Dual` by default.
All LiDARs will operate in `Dual (Strongest + Last)` mode. 

## Changes to Radar
1. Modified `radar.launch.xml` to change the topic remapping about twist (twist) and kinematic_state (odometry).

(~~It was originally remapped on the wrong topic.~~  )
**Originally some topics were accidentally contaminated.**
To resolve this issue, I added `<group>` to separate the scopes and ensure the correct topics are passed.

> [!CAUTION]
> * [`continental_launch_all_hw.xml` ](https://github.com/tier4/nebula/blob/main/nebula_ros/src/continental/continental_ars548_hw_interface_wrapper.cpp#L54-L55)requires `TwistWithCovarianceStamped` as the `odometry_input`. :melting_face: 
> * [`radar_tracks_msgs_converter.launch.xml`](https://github.com/tier4/autoware.universe/blob/tier4/main/perception/autoware_radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node.cpp#L92-L93) requires `Odometry` as `input/odometry`. :melting_face: 




## Changes to GNSS

1. Added septentrio_heading_converter.py.
This scripts converts the `atteuler` (attitude in Euler angles) topic output by septentrio_driver to `autoware_orientation`.
[TIER IV INTERNAL LINK](https://star4.slack.com/archives/CEV8XMJBV/p1732860210079259?thread_ts=1732860142.860499&cid=CEV8XMJBV)
The autoware_orientation topic is required by [gnss_poser](https://github.com/autowarefoundation/autoware.universe/blob/main/sensing/autoware_gnss_poser/README.md) to specify GNSS orientation. 

2. Removed unnecessary entries from `gnss.launch.xml`
* Deleted entries about the u-blox GNSS.
* Added septentrio_heading_converter.py.

3.  Updated `mosaic_x5_rover.param.yaml`
* Configured to output attitude topic.
* Adjusted the heading offset to accommodate the antenna configuration in XX1 Gen2. [TIER IV INTERNAL LINK](https://tier4.atlassian.net/wiki/spaces/AIP/pages/3300819405/XX1+Gen2+GNSS+septentrio+AsteRx+SB3+Pro#%E5%8F%97%E4%BF%A1%E6%A9%9F%E6%9C%AC%E4%BD%93%E3%81%AE%E8%A8%AD%E5%AE%9A)

Commit:
update gnss settings for septentrio

Signed-off-by: Kento Yabuuchi <kento.yabuuchi.2@tier4.jp>

Commit:
set lidar config

Commit:
set att_offset for septentio

Commit:
use odometry for radar

Commit:
fix bug for radar twist/odometry topic contamination

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

KYabuuchi committed:
ignore some spell

Signed-off-by: Kento Yabuuchi <kento.yabuuchi.2@tier4.jp>

KYabuuchi committed:
fix: autoware prefix

Signed-off-by: Kento Yabuuchi <kento.yabuuchi.2@tier4.jp>

shmpwk commented on aip_xx1_gen2_launch/launch/radar.launch.xml:
```diff
@@ -25,7 +25,7 @@
 
   <group if="$(eval &quot;'$(var ars_version)'=='ars548'&quot;)">
-    <let name="odometry_topic" value="/sensing/vehicle_velocity_converter/twist_with_covariance"/>
+    <let name="odometry_topic" value="/localization/kinematic_state"/>
```
```suggestion
```
I haven't checked the all configs but as I remember each default `odometry_topic` is `/localization/kinematic_state` ...? Any exception?

KYabuuchi replied:
No, this radar.launxh.xml calls [continental_launch_all_hw.xml in Nebula](https://github.com/tier4/nebula/blob/main/nebula_ros/launch/continental_launch_all_hw.xml#L13), but the odometry_topic is set to a meaningless topic by default in that nebula file. 

Therefore, we need to specify the correct odometry_topic here.

shmpwk replied:
Thank you, make sense!

shmpwk reviewed (dismissed):
Wonderful! Please add type hinting for the python script.

KYabuuchi committed:
add type annotation in septentrio_heading_converter.py

Signed-off-by: Kento Yabuuchi <kento.yabuuchi.2@tier4.jp>

KYabuuchi commented:
@shmpwk  Hi! I added type hint to python script in 533a502

KYabuuchi commented:
When I verified this PR in the actual vehicle, I realized that further modifications to `radar.launch.xml` were necessary. :bow:
I have implemented the additional changes in ca1594c, and updated the PR description accordingly.

shmpwk reviewed (approved):
LGTM
